tools/warning-to-plist/warning_to_plist/converter/report.py

```diff
@@ -0,0 +1,134 @@
+    calculates the new line location.
+    Returns the line content without white spaces and the new column number.
+    E.g.:
+    line_content = "  int foo = 17;   sizeof(43);  "
+                                      ^
+                                      |- bug_col = 18
+    content_begin = "  int foo = 17;   "
+    content_begin_strip = "intfoo=17;"
+    line_strip_len = 18 - 10 => 8
+    ''.join(line_content.split()) => "intfoo=17;sizeof(43);"
+                                                ^
+                                                |- until_col - line_strip_len
+                                                       18    -     8
+                                                             = 10
+    """
+    content_begin = line_content[:old_col]
+    content_begin_strip = u''.join(content_begin.split())
+    line_strip_len = len(content_begin) - len(content_begin_strip)
+
+    return ''.join(line_content.split()), \
+           old_col - line_strip_len
+
+
+def generate_report_hash(main_section, source_file):
+    """ Generate unique hashes for reports.
+
+    Hash generation algoritm for older plist versions where no issue hash was
+    generated or for the plists generated where the issue hash generation
+    feature is still missing.
+
+    High level overview of the hash content:
+     * file_name from the main diag section
+     * checker message
+     * line content from the source file if can be read up
+     * column numbers from the main diag sections location
+     * all the whitespaces from the source content are removed
+
+    """
+
+    try:
+        m_loc = main_section.get('location')
+        source_line = m_loc.get('line')
+
+        from_col = m_loc.get('col')
+        until_col = m_loc.get('col')
+
+        # WARNING!!! Changing the error handling type for encoding errors
+        # can influence the hash content!
+        line_content = get_line(source_file, source_line, errors='ignore')
+
+        # Remove whitespaces so the hash will be independet of the
+        # source code indentation.
+        line_content, new_col = \
+            remove_whitespace(line_content, from_col)
+        # Update the column number in sync with the
+        # removed whitespaces.
+        until_col = until_col - (from_col-new_col)
+        from_col = new_col
+
+        if line_content == '' and not os.path.isfile(source_file):
+            LOG.error("Failed to include soruce line in the report hash.")
+            LOG.error('%s does not exists!', source_file)
+
+        file_name = os.path.basename(source_file)
+        msg = main_section.get('description')
+
+        hash_content = [file_name,
+                        msg,
+                        line_content,
+                        str(from_col),
+                        str(until_col)]
+
+        string_to_hash = '|||'.join(hash_content)
+        return hashlib.md5(string_to_hash.encode()).hexdigest()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of insecure MD2, MD4, MD5, or SHA1 hash function.](https://app.codacy.com/app/CodeChecker/codechecker/pullRequest?prid=4169881)